packages/massa-web3/src/experimental/basicElements/mas.ts

```diff
@@ -100,25 +100,33 @@ export function fromString(value: string): Mas {
   return fromNumber(mas)
 }
 
-/**
- * Converts a value in the smallest unit of the Massa currency to a decimal value.
+/*
+ * Converts a Mas value to a string.
+ *
+ * @param value - The Mas value.
+ * @param decimalPlaces - The number of decimal places to include in the string.
+ * @returns The value as a string.
  *
- * @param value - The value in the smallest unit of the Massa currency.
- * @param keepTrailingZeros - Optional parameter to keep trailing zeros in the decimal part.
- * @returns The decimal value.
+ * @throws An error if the value is too large to be represented by an U256.
  */
-export function toString(value: Mas, keepTrailingZeros = false): string {
+export function toString(
+  value: Mas,
+  decimalPlaces: number | null = null
+): string {
   if (value >= BigInt(ONE) << BigInt(SIZE_U256_BIT)) {
     throw new Error(ERROR_VALUE_TOO_LARGE)
   }
 
   const valueString = value.toString()
-  const integerPart = valueString.slice(FIRST, -NB_DECIMALS) || '0'
-
+  const integerPart = valueString.slice(ZERO, -NB_DECIMALS) || '0'
   let decimalPart = valueString.slice(-NB_DECIMALS).padStart(NB_DECIMALS, '0')
 
-  if (!keepTrailingZeros) {
+  if (decimalPlaces === null) {
     decimalPart = decimalPart.replace(/0+$/, '')
+  } else {
+    // TODO - Round the number to the desired number of decimal places
```

Comment:
Changing from truncate to round approximation is indeed a breaking change. Here's how you could handle it:

- If you plan to offer both options, you should add a proper argument to choose between rounding and truncating. If you don't want to spend time on the round implementation, simply throw a `not implemented` exception when it's used.
- If you decide to only offer round approximation, you should clearly document and implement it immediately.
- If you choose to only offer truncate approximation, you should document this  and remove the related TODO comment.